oneandone/src/main/java/org/apache/jclouds/oneandone/rest/domain/SingleServerAppliance.java

```diff
@@ -70,7 +70,63 @@
       "min_hdd_size", "type", "state", "version", "categories", "eula_url"})
    public static SingleServerAppliance create(String id, String name, List<AvailableDataCenters> availableDataCenters, String osInstallationBase, Types.OSFamliyType osFamily, String os,
            String osVersion, int osArchitecture, Types.OSImageType osImageType, int minHddSize, Types.ApplianceType type, String state, String version, List<String> categories, String eulaUrl) {
-      return new AutoValue_SingleServerAppliance(id, name, availableDataCenters == null ? ImmutableList.<AvailableDataCenters>of() : availableDataCenters, osInstallationBase, osFamily, os, osVersion, osArchitecture, osImageType, minHddSize, type, state, version, categories == null ? ImmutableList.<String>of() : categories, eulaUrl);
+      return builder()
+              .id(id)
+              .name(name)
+              .availableDataCenters(availableDataCenters == null ? ImmutableList.<AvailableDataCenters>of() : ImmutableList.copyOf(availableDataCenters))
+              .osInstallationBase(osInstallationBase)
+              .osFamily(osFamily)
+              .os(os)
+              .osVersion(osVersion)
+              .osArchitecture(osArchitecture)
+              .osImageType(osImageType)
+              .minHddSize(minHddSize)
+              .type(type)
+              .state(state)
+              .version(version)
+              .categories(categories == null ? ImmutableList.<String>of() : ImmutableList.copyOf(categories))
+              .eulaUrl(eulaUrl)
+              .build();
+   }
+
+   public static Builder builder() {
+      return new AutoValue_SingleServerAppliance.Builder();
+   }
+
+   @AutoValue.Builder
+   public abstract static class Builder {
+
+      public abstract Builder id(String id);
+
+      public abstract Builder name(String name);
+
+      public abstract Builder availableDataCenters(List<AvailableDataCenters> availableDataCenters);
+
+      public abstract Builder osInstallationBase(String osInstallationBase);
+
+      public abstract Builder osFamily(Types.OSFamliyType osFamily);
+
+      public abstract Builder os(String os);
+
+      public abstract Builder osVersion(String osVersion);
+
+      public abstract Builder osArchitecture(int osArchitecture);
+
+      public abstract Builder osImageType(Types.OSImageType osImageType);
+
+      public abstract Builder minHddSize(int minHddSize);
+
+      public abstract Builder type(Types.ApplianceType type);
+
+      public abstract Builder state(String state);
+
+      public abstract Builder version(String version);
+
+      public abstract Builder categories(List<String> categories);
+
+      public abstract Builder eulaUrl(String eulaUrl);
+
+      public abstract SingleServerAppliance build();
```

Comment:
Use [this pattern](https://github.com/jclouds/jclouds-labs/blob/master/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/AvailabilitySet.java#L82-L90) to enforce immutable collections in builders.